src/App.tsx

```diff
@@ -14,11 +15,116 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[],
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  buttonAction = (good: string): void => {
+    if (!this.state.selectedGoods.includes(good)) {
+      this.setState((state) => {
+        state.selectedGoods.push(good);
+
+        return { selectedGoods: state.selectedGoods };
+      });
+    } else {
+      this.setState((state) => {
+        const newList = state.selectedGoods.filter(item => item !== good);
+
+        return { selectedGoods: newList };
+      });
+    }
+  };
+
+  messageLine = (list: string[] | []) => {
+    if (list.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (list.length === 1) {
+      return `${list.join(', ')} is selected`;
+    }
+
+    const result = list.join(', ');
+    const index = result.lastIndexOf(', ');
+
+    return `${result.slice(0, index)} and ${result.slice(index + 2)} are selected`;
+  };
+
+  clearButton = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  render(): React.ReactNode {
+    const { selectedGoods } = this.state;
+
+    const selectedOrNot = (checkItem: string): boolean => {
+      return selectedGoods.includes(checkItem);
+    };
+
+    return (
+      <div className="block container container--margin">
+        <div className="box is-size-2 container--margin">
+          <h1>
+            {this.messageLine(selectedGoods)}
+          </h1>
+          <button
+            type="button"
+            onClick={this.clearButton}
+            className={cn(
+              'button',
+              'is-light',
+              'is-danger',
+              { 'is-invisible': (selectedGoods.length === 0) },
+            )}
+          >
+            Clear
+          </button>
+        </div>
+        <ul
+          className="
+          columns
+          "
```

Comment:
it looks not good. can you write thr only className in one line?

Reply:
My fail! In previase, there were several class for more styles but I've deleted them!